src/processing/getFiatRate/utils.ts

```diff
@@ -1,25 +1,34 @@
-import { AxiosResponse } from 'axios';
 import { ITrade } from '../../types';
 
-interface ICryptoCompareResponse {
-    [key: string]: number;
+export function sleep(ms: number, limit: string) {
+    console.warn(`Rate limit exceeded for the ${limit}, retrying at the start of next ${limit}, in ${ms} ms...`);
+    return new Promise(resolve => setTimeout(resolve, ms));
 }
 
-export function cryptocompareRateResponse(response: AxiosResponse<ICryptoCompareResponse>, fiatCurrency: string) {
-    if ('data' in response) {
-        try {
-            const result = response.data;
-            if (result[fiatCurrency] !== 0) {
-                return result[fiatCurrency];
-            } else {
-                return false;
-            }
-        } catch (ex) {
-            throw new Error('Error parsing JSON');
-        }
-    } else {
-        throw new Error('Invalid Response');
+export function waitForRateLimit(rateLimit: any) {
+    let now = new Date();
+    let ms = 1000 - now.getMilliseconds();
+    if (rateLimit.calls_made.second > rateLimit.max_calls.second) {
+        return sleep (ms, 'second');
+    }
+    ms += (60 - now.getSeconds() - 1) * 1000;
+    if (rateLimit.calls_made.minute > rateLimit.max_calls.minute) {
+        return sleep (ms, 'minute');
+    }
+    ms += (60 - now.getMinutes() - 1) * 60 * 1000;
+    if (rateLimit.calls_made.hour > rateLimit.max_calls.hour) {
+        return sleep (ms, 'hour');
+    }
+    ms += (24 - now.getHours() - 1) * 60 * 60 * 1000;
+    if (rateLimit.calls_made.day > rateLimit.max_calls.day) {
+        return sleep (ms, 'day');
+    }
+    let numberOfDaysInTheMonth = new Date(now.getFullYear(), now.getMonth(), 0).getDate();
+    ms += (numberOfDaysInTheMonth - now.getDate() - 1) * 24 * 60 * 60 * 1000;
+    if (rateLimit.calls_made.month > rateLimit.max_calls.month) {
+        return sleep (ms, 'month');
```

Comment:
Awesome function, but I feel like no one is going to wait a month or even probably a day to let this keep processing. Even an hour Id probably just throw an error and say something like rate limit hit try again later. Or if we want to be extra informative can tell them they have to wait a few days, month, etc..

Reply:
Yeah thats a good idea, if/when you do that I would also suggest planning for multiple api keys as I can imagine someone wanting to use multiple keys to avoid hitting limits

Reply:
That would be great indeed

Reply:
So I’m throwing an error. In the case of trades import, I’m showing the message in AlertData.
The api key setting can be done separately

Reply:
See in function sleep, we don’t wait for months/days/hours. We throw an error. That error is catched in [pages/import.tsx](https://github.com/starsoccer/cryptotithe/pull/276/files#diff-7355506b1cb0bc25b4c151ec69be92c9f154e3c990f811ffe1a90d95c5e742ae) and displayed to the user

Reply:
![image](https://user-images.githubusercontent.com/10415894/207625694-89b5e860-b023-44c5-8959-7e5c7cddfd2f.png)